### PR TITLE
Refactor WorkSpaces resources to use keyvaluetags package

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #10688.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesIpGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesIpGroup_ -timeout 120m
=== RUN   TestAccAwsWorkspacesIpGroup_basic
=== PAUSE TestAccAwsWorkspacesIpGroup_basic
=== CONT  TestAccAwsWorkspacesIpGroup_basic
--- PASS: TestAccAwsWorkspacesIpGroup_basic (44.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	44.423s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesDirectory'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesDirectory -timeout 120m
=== RUN   TestAccAwsWorkspacesDirectory
=== RUN   TestAccAwsWorkspacesDirectory/basic
=== RUN   TestAccAwsWorkspacesDirectory/subnetIds
--- PASS: TestAccAwsWorkspacesDirectory (1464.50s)
    --- PASS: TestAccAwsWorkspacesDirectory/basic (852.48s)
    --- PASS: TestAccAwsWorkspacesDirectory/subnetIds (612.02s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1464.541s
```

* Remove the generated version of `WorkspacesUpdateTags` as it leads to the removal of existing tags that are unchanged
~* Move the [custom version of `WorkspacesUpdateTags`](https://github.com/terraform-providers/terraform-provider-aws/pull/11023#issuecomment-568214709) to the `keyvaluetags/custom` directory~
~* Includes commit from https://github.com/terraform-providers/terraform-provider-aws/pull/11638~

### CODE REVIEW

**Comment by ewbankkit** [comment]
#### TODO

WorkSpaces workspaces (https://github.com/terraform-providers/terraform-provider-aws/pull/11608) can be managed by [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-workspaces-workspace.html).
See if we need to handle system tags like we did for https://github.com/terraform-providers/terraform-provider-aws/issues/7323, https://github.com/terraform-providers/terraform-provider-aws/pull/7342.

**Comment by ewbankkit** [comment]
Tested with [`ignore_tags`](https://github.com/terraform-providers/terraform-provider-aws/issues/10689) functionality to verify that tag updating works when there are ignore tags.

Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesIpGroup_ignoreTags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesIpGroup_ignoreTags -timeout 120m
=== RUN   TestAccAwsWorkspacesIpGroup_ignoreTags
=== PAUSE TestAccAwsWorkspacesIpGroup_ignoreTags
=== CONT  TestAccAwsWorkspacesIpGroup_ignoreTags
--- PASS: TestAccAwsWorkspacesIpGroup_ignoreTags (37.95s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	38.002s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesDirectory/ignoreTags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesDirectory/ignoreTags -timeout 120m
=== RUN   TestAccAwsWorkspacesDirectory
=== RUN   TestAccAwsWorkspacesDirectory/ignoreTags
--- PASS: TestAccAwsWorkspacesDirectory (947.18s)
    --- PASS: TestAccAwsWorkspacesDirectory/ignoreTags (947.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	947.249s
```

File: aws/resource_aws_workspaces_directory.go [hunk 1]
```diff
@@ -140,6 +140,8 @@ func resourceAwsWorkspacesDirectoryCreate(d *schema.ResourceData, meta interface
 
 func resourceAwsWorkspacesDirectoryRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).workspacesconn
+	ignoreTags := meta.(*AWSClient).ignoreTags
```
Comment (bflad) [inline aws/resource_aws_workspaces_directory.go]: Hi @ewbankkit 👋 We would prefer to not add resources/logic to the `ignore_tags` preview at this time. We are finishing up a round of feedback on the functionality, so the implementation may change.

Comment (ewbankkit) [reply aws/resource_aws_workspaces_directory.go]: @bflad I was using the `ignore_tags` functionality to test the claim of the [`CreateTags`](https://docs.aws.amazon.com/workspaces/latest/api/API_CreateTags.html) API:

> If you want to add new tags to a set of existing tags, you must submit all of the existing tags along with the new ones.

by having tags that are not managed by Terraform and seeing if they get removed when updating tags via Terraform - it turns out they don't so the API in fact doesn't overwrite any existing tags.
I'll change this PR to _WIP_ and remove the `ignore_tags` stuff.
I'll use the same approach to test the S3 object tagging functionality in https://github.com/terraform-providers/terraform-provider-aws/pull/11964 but again not leave the changes in the final PR.

**Comment by ewbankkit** [comment]
OK, removed the `ignore_tags` changes as mentioned in the review comments above.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesIpGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesIpGroup_ -timeout 120m
=== RUN   TestAccAwsWorkspacesIpGroup_basic
=== PAUSE TestAccAwsWorkspacesIpGroup_basic
=== CONT  TestAccAwsWorkspacesIpGroup_basic
--- PASS: TestAccAwsWorkspacesIpGroup_basic (44.64s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	44.684s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesDirectory'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesDirectory -timeout 120m
=== RUN   TestAccAwsWorkspacesDirectory
=== RUN   TestAccAwsWorkspacesDirectory/basic
=== RUN   TestAccAwsWorkspacesDirectory/subnetIds
--- PASS: TestAccAwsWorkspacesDirectory (1398.14s)
    --- PASS: TestAccAwsWorkspacesDirectory/basic (788.30s)
    --- PASS: TestAccAwsWorkspacesDirectory/subnetIds (609.84s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1398.189s
```

Removing _WIP_.

**Comment by ewbankkit** [comment]
Rebased to remove merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesIpGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesIpGroup_ -timeout 120m
=== RUN   TestAccAwsWorkspacesIpGroup_basic
=== PAUSE TestAccAwsWorkspacesIpGroup_basic
=== CONT  TestAccAwsWorkspacesIpGroup_basic
--- PASS: TestAccAwsWorkspacesIpGroup_basic (45.06s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	45.101s
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAwsWorkspacesDirectory'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAwsWorkspacesDirectory -timeout 120m
=== RUN   TestAccAwsWorkspacesDirectory
=== RUN   TestAccAwsWorkspacesDirectory/basic
=== RUN   TestAccAwsWorkspacesDirectory/disappears
=== RUN   TestAccAwsWorkspacesDirectory/subnetIds
--- PASS: TestAccAwsWorkspacesDirectory (1847.66s)
    --- PASS: TestAccAwsWorkspacesDirectory/basic (651.98s)
    --- PASS: TestAccAwsWorkspacesDirectory/disappears (596.50s)
    --- PASS: TestAccAwsWorkspacesDirectory/subnetIds (599.17s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1847.706s
```

**Comment by ewbankkit** [comment]
Rebased to include https://github.com/terraform-providers/terraform-provider-aws/commit/ef098db34234c4f533be2a3f5063bf609e960337 and pass linting.

**Pull request merged**